DapperDoodle/BaseSqlExecutor.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Data;
+using Dapper;
+using DapperDoodle.Interfaces;
+
+namespace DapperDoodle
+{
+    public class BaseSqlExecutor
+    {
+        public void InitialiseDependencies(IBaseSqlExecutorOptions options)
+        {
+            _connection = options.Connection;
+            Dbms = options.Dbms;
+        }
+
+        private IDbConnection _connection;
+        public DBMS Dbms { get; set; }
+
+        public T SelectQuery<T>(string sql, object parameters = null)
+        {
+            if (typeof(T) == typeof(int))
+            {
+                return _connection.QueryFirst<T>(sql, parameters);
+            }
+            
+            return (T)_connection.Query<T>(sql, parameters);
+        }
+
+        public IDbConnection GetIDbConnection()
+        {
+            return _connection;
+        }
+
+        protected int Execute(string sql, object parameters = null)
+        {
+            if (sql.Equals("", StringComparison.InvariantCulture) || sql is null)
```

Comment:
you might want to swap these around - check for null firs, and then do .Equals

OTOH, use `string.IsNullOrWhitespace` which will also cater for other white space strings

Reply:
Thanks @fluffynuts. I will do that. 👍🏼